README.md

```diff
@@ -0,0 +1,82 @@
+# Chief
+
+A simple command pattern for Ruby.
+
+[![Code Climate](https://codeclimate.com/github/policygenius/chief/badges/gpa.svg)](https://codeclimate.com/github/policygenius/chief)
+
+
+## Usage
+
+A `Chief::Command`'s primary interface is the `.call` method, executes the command and returns a `Chief::Result` instance.
+
+A `Chief::Result` encapsulates whether the command succeeding, including the return values and (optionally) any errors encountered.
+
+Suppose we have an ExampleApplication, that needs to notify an external service with a message. First, we create the command:
+
+```Ruby
+require 'net/http'
+
+module ExampleApplication
+  module SomeExternalService
+    class Notify < Chief::Command
+      attr_reader :message
+
+      def initialize(message)
+        @message = message
+      end
+
+      def call
+        response = report_to_external_service
+
+        if response.code == 200
+          success!(message)
+        else
+          fail!(false, 'Could not notify the external service')
```

Comment:
Most times I fail, I found that I don't need the value back, thus it would be nice if we could do:

`fail!` # sometimes I just want to fail without a message or value
`fail! 'Could not notify the external service'`
`fail! 'Unexpected error while notifying service', exception`

I assume the value is first though to be consistent with `success!`?